src/LogFileReader.cpp

```diff
@@ -47,12 +49,44 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    unsigned int allowedCharLength = 15;
+    std::string allowedChars [allowedCharLength] = {"1","2","3","4","5","6","7","8","9","0",":",",",".","-"," "};
+    bool tempBool;
+    for(unsigned int count = 0; count < line.toStdString().length(); count++)
+    {
+        tempBool = false;
+        for(unsigned int count1 = 0; count1 < allowedCharLength; count1++)
+        {
+            if(allowedChars[count1] == line.toStdString().substr(count,1))
+            {
+                tempBool = true;
+            }
+        }
+        if(tempBool == false)
+        {
+            return false;
+        }
+    }
+
+    //example input= 07:01:07.024, 3.538654, 25.523571
+    QStringList lineSplit = line.split(",");
+    if(lineSplit.length()!= 3)
+        {return false;}
+    QStringList timeSplit = lineSplit[0].split(":");
+    if(timeSplit.length()!=3)
+        {return false;}
+    QStringList secondSplit = timeSplit[2].split(".");
+    if(secondSplit.length()!=2)
+        {return false;}
+
+    QTime time = QTime(timeSplit[0].toInt(),    //hour
+                        timeSplit[1].toInt(),   //minute
+                        secondSplit[0].toInt(), //second
+                        secondSplit[1].toInt());//millisec
```

Comment:
Instead of building the time manually, I would recommend using the built in function to convert a string into a QTime http://doc.qt.io/qt-5/qtime.html#fromString-1